### PR TITLE
Removed code thanks to Equatable/Hashable synthesis

### PR DESCRIPTION
**Pull request by hartbit**

### CODE REVIEW

**Comment by neonichu** [comment]
@swift-ci please smoke test

File: Sources/Utility/Version.swift [hunk 1]
```diff
@@ -45,16 +45,7 @@ public struct Version {
...
-    static public func == (lhs: Version, rhs: Version) -> Bool {
-        return lhs.major == rhs.major &&
-               lhs.minor == rhs.minor &&
-               lhs.patch == rhs.patch &&
-               lhs.prereleaseIdentifiers == rhs.prereleaseIdentifiers &&
-               lhs.buildMetadataIdentifiers == rhs.buildMetadataIdentifiers
-    }
-
+extension Version {
     public var hashValue: Int {
```
Comment (aciidb0mb3r) [inline Sources/Utility/Version.swift]: `hashValue` should also get synthesized, right?

File: Sources/Basic/ByteString.swift [hunk 2]
```diff
@@ -91,7 +91,7 @@ extension ByteString: CustomStringConvertible {
 }
 
 /// Hashable conformance for a ByteString.
-extension ByteString: Hashable {
+extension ByteString {
```
Comment (aciidb0mb3r) [inline Sources/Basic/ByteString.swift]: Why move the conformance to struct decl?

**Comment by hartbit** [comment]
@swift-ci please smoke test

File: Sources/PackageDescription/Version.swift [hunk 3]
```diff
@@ -49,10 +49,9 @@ extension Version: Equatable {
 
 // MARK: Hashable
 
+//FIXME: Remove in 4.2
```
Comment (aciidb0mb3r) [inline Sources/PackageDescription/Version.swift]: This can be done now because PackageDescription does not need to be 4.0 compatible.

**Pull request merged**